internal/generator/vector/output/splunk/splunk.go

```diff
@@ -0,0 +1,129 @@
+	genhelper "github.com/openshift/cluster-logging-operator/internal/generator/helpers"
+	. "github.com/openshift/cluster-logging-operator/internal/generator/vector/elements"
+	vectorhelpers "github.com/openshift/cluster-logging-operator/internal/generator/vector/helpers"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	splunkEncodingJson = fmt.Sprintf("%q", "json")
+)
+
+type Splunk struct {
+	ComponentID  string
+	Inputs       string
+	Endpoint     string
+	DefaultToken string
+}
+
+func (s Splunk) Name() string {
+	return "SplunkVectorTemplate"
+}
+
+func (s Splunk) Template() string {
+	return `{{define "` + s.Name() + `" -}}
+[sinks.{{.ComponentID}}]
+type = "splunk_hec"
+inputs = {{.Inputs}}
+endpoint = "{{.Endpoint}}"
+compression = "none"
+default_token = "{{.DefaultToken}}"
+{{end}}`
+}
+
+type SplunkEncoding struct {
+	ComponentID string
+	Codec       string
+}
+
+func (se SplunkEncoding) Name() string {
+	return "splunkEncoding"
+}
+
+func (se SplunkEncoding) Template() string {
+	return `{{define "` + se.Name() + `" -}}
+[sinks.{{.ComponentID}}.encoding]
+codec = {{.Codec}}
+{{end}}`
+}
+
+func Conf(o logging.OutputSpec, inputs []string, secret *corev1.Secret, op Options) []Element {
+	if genhelper.IsDebugOutput(op) {
+		return []Element{
+			Debug(strings.ToLower(vectorhelpers.Replacer.Replace(o.Name)), vectorhelpers.MakeInputs(inputs...)),
+		}
+	}
+	return MergeElements(
+		[]Element{
+			Output(o, inputs, secret, op),
+			Encoding(o),
+		},
+		TLSConf(o, secret),
+	)
+}
+
+func Output(o logging.OutputSpec, inputs []string, secret *corev1.Secret, op Options) Element {
+	return Splunk{
+		ComponentID:  strings.ToLower(vectorhelpers.Replacer.Replace(o.Name)),
+		Inputs:       vectorhelpers.MakeInputs(inputs...),
+		Endpoint:     o.URL,
+		DefaultToken: security.GetFromSecret(secret, "hecToken"),
+	}
+}
+
+func Encoding(o logging.OutputSpec) Element {
+	return SplunkEncoding{
+		ComponentID: strings.ToLower(vectorhelpers.Replacer.Replace(o.Name)),
+		Codec:       splunkEncodingJson,
+	}
+}
+
+func TLSConf(o logging.OutputSpec, secret *corev1.Secret) []Element {
+	conf := []Element{}
+	if o.Secret != nil {
+		hasTLS := false
+		conf = append(conf, security.TLSConf{
+			ComponentID:        strings.ToLower(vectorhelpers.Replacer.Replace(o.Name)),
+			InsecureSkipVerify: o.TLS != nil && o.TLS.InsecureSkipVerify,
+		})
+
+		if o.Name == logging.OutputNameDefault || security.HasTLSCertAndKey(secret) {
+			hasTLS = true
+			kc := TLSKeyCert{
+				CertPath: security.SecretPath(o.Secret.Name, constants.ClientCertKey),
+				KeyPath:  security.SecretPath(o.Secret.Name, constants.ClientPrivateKey),
+			}
+			conf = append(conf, kc)
+		}
+		if o.Name == logging.OutputNameDefault || security.HasCABundle(secret) {
+			hasTLS = true
+			ca := CAFile{
+				CAFilePath: security.SecretPath(o.Secret.Name, constants.TrustedCABundleKey),
+			}
+			conf = append(conf, ca)
+		}
+		if !hasTLS {
+			return []Element{}
+		}
+	} else if secret != nil {
```

Comment:
dont think this is required either

Reply:
We really need to centralize this TLS logic somewhere. 
@xperimental did you make any progress on that?

Reply:
Look like a lot of duplication of TLS logic code, i did similar to Kafka config

Reply:
> did you make any progress on that?

Sadly, no. The plan was to move the logic into the `security` package, so that outputs using the standard TLS options can benefit from de-duplication.

The kafka output is not a good example, because it does not actually use the standard TLS configuration of vector. I did not notice that in the original implementation, but my fix (#1591) is still not merged. The `splink_hec` output does seem to use the standard TLS options, so you could start consolidating the TLS options in for example Elasticsearch and Splunk to get to a reusable package.

Another change i did in the linked PR is to remove the `hasTLS` flag in the code and not make TLS dependent on the existence of a "secret" (passphrase, CA, certificate) but instead only on the used URL. I think this makes it more transparent to the user.